control-plane/subcommand/inject-connect/v1controllers.go

```diff
@@ -316,64 +315,60 @@ func (c *Command) configureV1Controllers(ctx context.Context, mgr manager.Manage
 			return err
 		}
 
-		mgr.GetWebhookServer().Register("/mutate-v1alpha1-peeringacceptors",
-			&ctrlRuntimeWebhook.Admission{Handler: &v1alpha1.PeeringAcceptorWebhook{
-				Client: mgr.GetClient(),
-				Logger: ctrl.Log.WithName("webhooks").WithName("peering-acceptor"),
-			}})
-		mgr.GetWebhookServer().Register("/mutate-v1alpha1-peeringdialers",
-			&ctrlRuntimeWebhook.Admission{Handler: &v1alpha1.PeeringDialerWebhook{
-				Client: mgr.GetClient(),
-				Logger: ctrl.Log.WithName("webhooks").WithName("peering-dialer"),
-			}})
+		(&v1alpha1.PeeringAcceptorWebhook{
+			Client: mgr.GetClient(),
+			Logger: ctrl.Log.WithName("webhooks").WithName("peering-acceptor"),
+		}).SetupWithManager(mgr)
```

Comment:
You'll see this pattern throughout `v1controllers.go` and `v2controllers.go`.

The gist is that a `Decoder` is no longer injected by the manager automatically like it was previously -- see [controller-runtime release notes RE: removing dependency injection](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0) -- so we need to do more setup ourselves. Rather than do all that here, I decided to follow the same `SetupWithManager` pattern for webhooks that we already use when initializing our controllers.

Reply:
👨🏻‍🍳 👏🏻